File: 05. Sorting/01. Normal Sorting Algorithms/03. Selection Sort/SelectionSort.java
class SelectionSort {

	static void swap(int[] arr, int i, int j) {
		int tmp = arr[i];
		arr[i] = arr[j];
		arr[j] = tmp;
	}

	static void selectionSort(int[] arr, int n) {
		for (int i=0; i<n; i++) {
			int minIndex = i;
			for (int j=i+1; j<n; j++) {
				if (arr[j] < arr[minIndex]) {
					minIndex = j;
				}
			}
			if (minIndex != i) {
				swap(arr, minIndex, i);
			}
		}

		// printing sorted array
		for (int i=0; i<n; i++) {
			System.out.print(arr[i]+" ");
		}
		System.out.println();
	}

	public static void main(String[] args) {
		int[] arr = {13, 32, 5, 0, 34, 67};
		selectionSort(arr, arr.length);
	}
}

File: 05. Sorting/01. Normal Sorting Algorithms/03. Selection Sort/selectionSort.js
function swap(arr, i, j) {
  let tmp = arr[i];
  arr[i] = arr[j];
  arr[j] = tmp;
}

function selectionSort(arr, n) {
  for (let i = 0; i < n-1; i++) {
    var minIndex = i;
    for (let j=i+1; j<n; j++) {
      if (arr[minIndex] > arr[j]) {
        minIndex = j;
      }
    }
    if (minIndex != i) {
      swap(arr, minIndex, i);
    }
  }
}

var arr = [32,19,76,7,24];
selectionSort(arr, arr.length);
var s = "";
for (let i=0; i<arr.length; i++) {
  s += arr[i]+" ";
}
console.log(s);
